test/uri/test_generic.rb

```diff
@@ -159,6 +159,13 @@ def test_parse
     assert_equal(nil, url.userinfo)
   end
 
+  def test_parse_scheme_with_symbols
+    # Valid schemes from https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml
+    assert_equal 'ms-search', URI.parse('ms-search://localhost').scheme
+    assert_equal 'microsoft.windows.camera', URI.parse('microsoft.windows.camera://localhost').scheme
+    assert_equal 'coaps+ws', URI.parse('coaps+ws:localhost').scheme
+  end
```

Comment:
It'd be interesting to add a test for `register_scheme` with a `-`, `.` or `+`.
Probably ends up in an exception, but better than e.g. allowing to register but then the class is not used when parsing.

Reply:
Yeah, I wasn't sure if the intent was to have `register_scheme` be a public API or not. My priority was to fix the parsing of arbitrary schemes.

If register is public API, then I'd rather leave it to you to fix it.

Reply:
I"d consider it public API because it is documented in https://github.com/ruby/uri/blob/86de79c09b82e390dce343925d31d6beb2aaed67/lib/uri.rb#L33